core/src/main/scala/kafka/utils/CoreUtils.scala

```diff
@@ -252,16 +255,62 @@ object CoreUtils {
     listenerListToEndPoints(listeners, securityProtocolMap, true)
   }
 
+  def validateOneIsIpv4AndOtherIpv6(first: String, second: String): Boolean =
```

Comment:
nit: since methods exposed in `CoreUtils` are expected to be used by other instances, I don't think we expect the `validateOneIsIpv4AndOtherIpv6` or `checkDuplicateListenerPorts` to be used by other cases. Maybe we can move them into `listenerListToEndPoints` like what `validate` method do.

Reply:
Commit pushed.